koap/src/commonMain/kotlin/Message.kt

```diff
@@ -291,17 +293,21 @@ sealed class Message {
              * - `0` (register) adds the entry to the list, if not present;
              * - `1` (deregister) removes the entry from the list, if present.
              */
-            enum class Registration { Register, Deregister }
+            sealed class Registration {
+                object Register: Registration()
+                object Deregister: Registration()
+            }
 
             /**
              * Constructs an [Observe] to be included in a GET request.
              *
              * @see Registration
              */
+            @JsName("constructorWithRegistration")
             constructor(action: Registration) : this(
                 when (action) {
-                    Register -> 0L
-                    Deregister -> 1L
+                    Message.Option.Observe.Registration.Register -> 0L
+                    Message.Option.Observe.Registration.Deregister -> 1L
```

Comment:
Add appropriate imports so can be written as:

```suggestion
                    Register -> 0L
                    Deregister -> 1L
```